### PR TITLE
Reformat changelog release headings

### PR DESCRIPTION
**Pull request by straight-shoota**
Adds a page title as level one heading. Release headings move to level two heading. They're also reformatted to include a link to the GitHub release.
This is following the suggested format of https://common-changelog.org/#2-format

### CODE REVIEW

**Comment by HertzDevil** [comment]
What about `CHANGELOG.0.md`?

File: CHANGELOG.md [hunk 1]
```diff
@@ -507,16 +513,16 @@
...
 - Update shards 0.17.3 ([#13296](https://github.com/crystal-lang/crystal/pull/13296), thanks @straight-shoota)
 
-## Other
+### Other
 
 - Do not match expectations outside specs ([#13079](https://github.com/crystal-lang/crystal/pull/13079), thanks @HertzDevil)
 - Enable or fix specs that already work on Windows ([#13186](https://github.com/crystal-lang/crystal/pull/13186), thanks @HertzDevil)
 
-# 1.7.3
+## 1.7.3
```
Comment (beta-ziliani) [inline CHANGELOG.md:521]:
```suggestion
## [1.7.3] - 2023-03-07
[1.7.3]: https://github.com/crystal-lang/crystal/releases/1.7.3
```

**Review by beta-ziliani — COMMENTED** [review]
Spotted a missing link

File: CHANGELOG.md (hunk 1 above)
Comment (straight-shoota) [reply CHANGELOG.md:521]: Eagle eye 👁️

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,4 +1,7 @@
-# 1.9.0 (2023-07-11)
+# Changelog
+
+## [1.9.0] - 2023-07-11
+[1.9.0]: https://github.com/crystal-lang/crystal/releases/1.9.0
```
Comment (Sija) [inline CHANGELOG.md:4]: Readability suggestion:

```suggestion

[1.9.0]: https://github.com/crystal-lang/crystal/releases/1.9.0

```

ditto elsewhere

Comment (Fryguy) [inline CHANGELOG.md:4]: Out of curiosity why not just link it directly?

```suggestion
## [1.9.0](https://github.com/crystal-lang/crystal/releases/1.9.0) - 2023-07-11
```

Typically the reference-style link targets are all at the end of the document, like a footer, so I was surprised to see it immediately after where it's used.

**Pull request merged**